vsg/rules/align_tokens_in_region_between_tokens_unless_between_tokens.py

```diff
@@ -142,6 +145,19 @@ def analyze(self, oFile):
                            dAnalysis = {}
 
                    if self.case_control_statements_ends_group:
+                       if check_for_case_keywords(iIndex + 1, lTokens) or check_for_when_keywords(iIndex + 1, lTokens):
```

Comment:
Replace the condition in the if statement at line 148 with another function like `is_case_keyword`:
```python
148            if is_case_keyword(self.case_control_statement_ends_group, iIndex, lTokens):
```
Then create the function at the end of the file:
```python
def is_case_keyword(config, iIndex, lTokens):
    if check_for_case_keywords(iIndex + 1, lTokens):
        return True
    if check_for_when_keywords(iIndex + 1, lTokens):
        if config == 'break_at_whens':
           return True
    return False
```
I think that is about all you need to do.

I would update the test first to add the desired 3 values to the existing `case_control_statements_ends_group` before making any changes to this file.

Reply:
Done